src/wasm/thorvgwasm.cpp

```diff
@@ -0,0 +1,175 @@
+
+class __attribute__((visibility("default"))) ThorvgWasm
+{
+public:
+    static unique_ptr<ThorvgWasm> create()
+    {
+        return unique_ptr<ThorvgWasm>(new ThorvgWasm());
+    }
+
+    string getError()
+    {
+        return mErrorMsg;
+    }
+
+    bool load(string data, int width, int height)
+    {
+        mErrorMsg = "None";
+
+        if (!mSwCanvas) {
+            mErrorMsg = "Canvas is NULL";
+            return false;
+        }
+
+        mPicture = Picture::gen().release();
+        if (!mPicture) {
+            mErrorMsg = "Picture get failed";
+            return false;
+        }
+
+        mSwCanvas->clear();
+
+        if (data.empty()) data = defaultData;
+        const char *cdata = data.c_str();
+        if (mPicture->load(cdata, strlen(cdata)) != Result::Success) {
+
+            /* mPicture is not handled as unique_ptr yet, so delete here */
+            delete(mPicture);
+            mPicture = nullptr;
+
+            mErrorMsg = "Load failed";
+            return false;
+        }
+
+        getDefaultSize();
+        setScale(width, height);
+
+        if (mSwCanvas->push(unique_ptr<Picture>(mPicture)) != Result::Success) {
+            mErrorMsg = "Push failed";
+            return false;
+        }
+
+        return true;
+    }
+
+    void update(int width, int height)
+    {
+        mErrorMsg = "None";
+
+        if (!mSwCanvas) {
+            mErrorMsg = "Canvas is NULL";
+            return;
+        }
+
+        if (!mPicture) {
+            mErrorMsg = "Picture is NULL";
+            return;
+        }
+
+        if (mWidth == width && mHeight == height) {
+            mErrorMsg = "Size is same";
+            return;
+        }
+
+        setScale(width, height);
+
+        if (mSwCanvas->update(mPicture) != Result::Success) {
+            mErrorMsg = "Update failed";
+            return;
+        }
+
+        return;
+    }
+
+    val render()
+    {
+        mErrorMsg = "None";
+
+        if (!mSwCanvas) {
+            mErrorMsg = "Canvas is NULL";
+            return val(typed_memory_view<uint8_t>(0, nullptr));
+        }
+
+        if (mSwCanvas->draw() != Result::Success) {
+            mErrorMsg = "Draw failed";
+            return val(typed_memory_view<uint8_t>(0, nullptr));
+        }
+
+        mSwCanvas->sync();
+
+        return val(typed_memory_view(mWidth * mHeight * 4, mBuffer.get()));
+    }
+
+    float getScale()
+    {
+        return mScale;
+    }
+
+private:
+    explicit ThorvgWasm()
+    {
+        mErrorMsg = "None";
+
+        Initializer::init(CanvasEngine::Sw, 0);
+        mSwCanvas = SwCanvas::gen();
+        if (!mSwCanvas) {
+            mErrorMsg = "Canvas get failed";
+            return;
+        }
+    }
+
+    void getDefaultSize()
+    {
+        float w, h;
+        mPicture->viewbox(nullptr, nullptr, &w, &h);
+        mDefaultWidth = static_cast<uint32_t>(w);
+        mDefaultHeight = static_cast<uint32_t>(h);
+    }
+
+    void setScale(int width, int height)
```

Comment:
Does this necessary to realloc though size is same?

void setScale() {
if (size is same) return;
...
}

Reply:
done